### ISSUE TITLE
RK-M171S Unknown error occurred

### ISSUE THREAD

**Issue by c1tru55** [ClusterM/skykettle-ha#17]
Здравствуйте.

На последнем шаге долго крутится лоадер, потом показывает ошибку `Unknown error occurred`.

![](http://i.imgur.com/yms9kmL.png)

Чайник в режиме сопряжения, мак-адрес верный. В логах ничего не вижу, или не там смотрю.

Home Assistant Core 2022.6.6
Home Assistant Supervisor 2022.05.3
Home Assistant OS 8.2

**Comment by ClusterM** [comment]
Попробуйте версию из мастера, пожалуйста. И покажите лог: https://github.com/ClusterM/skykettle-ha/wiki/How-to-get-Home-Assistant-log

**Comment by c1tru55** [comment]
сам процесс добавления устройства очень нестабильный. на 10 попыток в лучшем случае только 1 успешное добавление устройства, но даже после добавления - на странице устройства все инпуты неактивны, я ничего не могу сделать.
![](http://i.imgur.com/XcT8Rdp.png)
также никакой реакции на включения/выключения на самом чайнике. плюс показывается очень много настроек, которых вообще нет у моего чайника, например подсветка. 

лог прикладываю.
[home-assistant.zip](https://github.com/ClusterM/skykettle-ha/files/9184588/home-assistant.zip)

> Попробуйте версию из мастера, пожалуйста.

я устанавливал через HACS, там обновлений никаких нет.

**Comment by ClusterM** [comment]
![image](https://user-images.githubusercontent.com/4236181/180870713-b659b2e6-c451-453e-a0d0-eb99f89bbdc7.png)

Сопряжение надо будет заново установить.

**Comment by c1tru55** [comment]
[home-assistant.zip](https://github.com/ClusterM/skykettle-ha/files/9188193/home-assistant.zip)
с мастер веткой без изменений, лог прикладываю

**Comment by ClusterM** [comment]
А не используются ли какие-нибудь другие интеграции работающие с Bluetooth?

**Comment by ClusterM** [comment]
И не работает ли в это время какое-то другое приложение/устройство?

**Comment by c1tru55** [comment]
> А не используются ли какие-нибудь другие интеграции работающие с Bluetooth?
только если device tracker
> И не работает ли в это время какое-то другое приложение/устройство?
для связи с чайником? нет, специально даже принудительно все позакрывал.

**Comment by ClusterM** [comment]
Пожалуйста, установите Home Assistant 2022.8.1 или новее и попробуйте версию из master. Она теперь работает через новую интеграцию "Bluetooth", так что убедитесь, что она включена и работает.

**Comment by c1tru55** [comment]
Обновился до последних версий, теперь страницы устройства выглядит намного лучше:
![](http://i.imgur.com/dFosfPs.png)
но включение/выключение по-прежнему не работает. либо после включения чекбокс секунд через 5 обратно анчекается, либо остается включенным, но на чайнике это никак не отображается.

**Comment by c1tru55** [comment]
хотя вот сейчас в очередной раз перезагрузил интеграцию, явно остановил приложение R4S (хотя в нем и не был подключен к чайнику) и смог его включить через эту интеграцию, после закипания состояние также изменилось автоматически. буду продолжать наблюдения.

**Comment by ClusterM** [comment]
> Обновился до последних версий, теперь страница устройства выглядит намного лучше

Используется ли постоянное подключение? Можно посмотреть логи Home Assistant?
https://github.com/ClusterM/skykettle-ha/wiki/How-to-get-Home-Assistant-log

Если есть возможность, то содержимое Bluetooth трафика тоже хотелось бы посмотреть: 
https://github.com/ClusterM/skykettle-ha/wiki/How-to-get-HCI-log-on-Android-device
Во время записи повключать разные режимы, разную температуру. А то поддержка этой модели весьма экспериментальная, мне её не на чем проверить.

**Comment by c1tru55** [comment]
> Используется ли постоянное подключение?

да. я же правильно понимаю, что это означает эксклюзивное использование именно чайника по блютусу, а не самого блютус адаптера?

> Во время записи повключать разные режимы, разную температуру. 

повключать где? на чайнике?

**Comment by pafftis** [comment]
У меня такая ошибка:
Не могу подключиться к чайнику. Пожалуйста, убедитесь, что он стоит на подставке, включен в розетку, утилите "hatttool" предоставлены права (смотрите инструкцию), и официальное приложение сейчас не подключено к чайнику.

**Comment by pafftis** [comment]
[ha.txt](https://github.com/ClusterM/skykettle-ha/files/9365463/ha.txt)
my log

**Comment by c1tru55** [comment]
Рано я радовался, на следующий день интеграция перестала работать, перезапуск интеграции/НА, смена интервала обновления/типа соединения не помогает. Логи прикладываю.
По блютус логу:
подключился через приложение к чайнику, было 90+ градусов. включил кипячение, потом на 96 выключил, но он сам остаточным теплом дошел до 100. потом долил холодной воды, температура упала до 80+ (а мб 60+ не помню уже), включил кипячение, он сам уже дошел до 100 градусов и выключился.

[btsnoop_hci.log](https://github.com/ClusterM/skykettle-ha/files/9389131/btsnoop_hci.log)
[home-assistant.log](https://github.com/ClusterM/skykettle-ha/files/9389132/home-assistant.log)

**Comment by ClusterM** [comment]
> По блютус логу:
> подключился через приложение к чайнику, было 90+ градусов. включил кипячение, потом на 96 выключил, но он сам остаточным теплом дошел до 100. потом долил холодной воды, температура упала до 80+ (а мб 60+ не помню уже), включил кипячение, он сам уже дошел до 100 градусов и выключился.
> 
> [btsnoop_hci.log](https://github.com/ClusterM/skykettle-ha/files/9389131/btsnoop_hci.log)
> [home-assistant.log](https://github.com/ClusterM/skykettle-ha/files/9389132/home-assistant.log)
> 
> 

Можете попробовать версию из master?

**Comment by SergeyPomelov** [comment]
For me - it's not even pairing most of times.

**Comment by c1tru55** [comment]
> Можете попробовать версию из master?

мастер версия не помогала. но после замены блютус-адаптера на Zexmte (Long Range) - все стало работать стабильно.

**Comment by rebelint** [comment]
> > Можете попробовать версию из master?
> 
> мастер версия не помогала. но после замены блютус-адаптера на Zexmte (Long Range) - все стало работать стабильно.

такая проблема. такой же адаптер и чайник. были проблемы при установке на linux? в usb вставлены другие устройства? zigbee например?

**Comment by c1tru55** [comment]
> такая проблема. такой же адаптер и чайник. были проблемы при установке на linux? в usb вставлены другие устройства? zigbee например?

в usb вставлен только адаптер (zigbee стик вставлен в другой порт). HA установлен в виртуальной машине.
